src/controller/eme-controller.js

```diff
@@ -231,35 +234,40 @@ class EMEController extends EventHandler {
     logger.log(`Access for key-system "${keySystem}" obtained`);
 
     const mediaKeysListItem = {
-      mediaKeys: null,
-      mediaKeysSession: null,
       mediaKeysSessionInitialized: false,
       mediaKeySystemAccess: mediaKeySystemAccess,
       mediaKeySystemDomain: keySystem
     };
 
     this._mediaKeysList.push(mediaKeysListItem);
 
-    mediaKeySystemAccess.createMediaKeys()
+    const mediaKeysPromise = Promise.resolve()
+      .then(() => {
+        return mediaKeySystemAccess.createMediaKeys();
+      })
       .then((mediaKeys) => {
+        const returnMediaKeys = () => {
+          this._onMediaKeysCreated();
+          return mediaKeys;
+        };
+
         logger.log(`Media-keys created for key-system "${keySystem}"`);
         mediaKeysListItem.mediaKeys = mediaKeys;
 
         // Using `MediaKeys.getStatusForPolicy()` to check available HDCP version,
         // only when you manually set up `minHdcpVersion` before.
-        if (
-          typeof this._minHdcpVersion !== 'undefined' &&
-          typeof mediaKeys.getStatusForPolicy === 'function'
-        ) {
+        if (typeof this._minHdcpVersion !== 'undefined' && typeof mediaKeys.getStatusForPolicy === 'function') {
           logger.log(`Checking accessbility of HDCP version ${this._minHdcpVersion}"`);
-          return mediaKeys.getStatusForPolicy({ minHdcpVersion: this._minHdcpVersion })
+          return mediaKeys
+            .getStatusForPolicy({ minHdcpVersion: this._minHdcpVersion })
             .then((status) => {
               if (status !== 'usable') {
-                return Promise.reject(new Error(`Not a valid HDCP policy status ${status}`));
+                throw new Error(`Not a valid HDCP policy status ${status}`);
```

Comment:
this will be handled by the next `.catch()` and throw an error log to the client